docs/reference/api/storage/FlashSimBlockDevice.md

```diff
@@ -0,0 +1,28 @@
+## FlashSimBlockDevice
+
+<span class="images">![](https://os-doc-builder.test.mbed.com/docs/development/mbed-os-api-doxy/class_slicing_block_device.png)<span>FlashSimBlockDevice class hierarchy</span></span>
+
+The FlashSimBlockDevice class is a block device adpator, whose purpose is to simulate the behavior of a flash component if the underlying block device doesn't support such a behavior. This includes the following:
+
+- Support the `erase` API (fills the erase unit with the predefined erase value).
+- Only allow programming of erased area or ones whose content is the same as the one given by the user.
```

Comment:
Query: What does this mean? Could you please elaborate?

Reply:
Do you refer to the "Only allow programming..." line? If so, it means the following: Other block devices allow unrestricted programming anywhere. Here we only allow programming of areas, which were previously erased or ones that contain the same data we want to program. This is just how Flash components behave.